pkg/util/validation/limits.go

```diff
@@ -144,6 +147,8 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.RulerMaxRulesPerRuleGroup, "ruler.max-rules-per-rule-group", 0, "Maximum number of rules per rule group per-tenant. 0 to disable.")
 	f.IntVar(&l.RulerMaxRuleGroupsPerTenant, "ruler.max-rule-groups-per-tenant", 0, "Maximum number of rule groups per-tenant. 0 to disable.")
 
+	f.DurationVar(&l.CompactorRetentionPeriod, "compactor.retention-period", 0, "Delete blocks containing samples older than the specified retention period. 0 to disable.")
```

Comment:
The CLI flag should match the YAML field name.

```suggestion
	f.DurationVar(&l.CompactorRetentionPeriod, "compactor.blocks-retention-period", 0, "Delete blocks containing samples older than the specified retention period. 0 to disable.")
```